packages/feature-detector/src/find-globals.ts

```diff
@@ -0,0 +1,52 @@
+import { dirname, resolve } from 'path'
+import { Project, SourceFile, ts } from 'ts-morph'
+import { buildProject } from './utils/project'
+
+/**
+ * Find the list of globals used by source files in the provided project.
+ * Analyzed source files can be filtered by provided a list of glob patterns (default to all TypeScript and JavaScript files, excluding type definitions)
+ */
+export function findGlobals(
+  sourcePath: string,
+  project: Project = buildProject()
+): string[] {
+  const globals = new Set<string>()
+  const sourceFile = project.getSourceFileOrThrow(sourcePath)
+  addFileGlobals(sourceFile, globals)
+  for (const imported of sourceFile.getImportDeclarations()) {
+    console.log(imported.getModuleSpecifierSourceFileOrThrow())
+    const importedFile = project.getSourceFile(
+      resolve(
+        dirname(sourceFile.getFilePath()),
+        imported.getModuleSpecifierValue()
+      )
+    )
```

Comment:
Are source files cached in `project` or will be parsed every time? I guess we'll integrate this with other packages using the AST.

Reply:
I am sorry. I accidentally shipped an experiment for analyzing imports. Unfortunately, this experiment does not work, so I'm reverting to what it was initially: analyzing a single file (it's supposed to be pre-bundled, so no require/import in it).

Reply:
About your questions: source files are cached into the project (as far as I know).
That's why I made it possible to pass the project as a parameter: so caller code may reuse an existing TS-morph project.

Reply:
> analyzing a single file (it's supposed to be pre-bundled, so no require/import in it).

Oh I didn't realize that. It makes 10000% sense! 👍